File: 0027-remove-element/0027-remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let valCounter = 0;
    newArr = []
    for (let i = 0; i < nums.length; i++){
        if( nums[i]!== val){
            newArr.push(nums[i])
        }
    }

    let k = newArr.length

    for(let i = 0; i < k; i++){
        nums[i] = newArr[i]
    }

    return k
};

File: 0027-remove-element/0027-remove-element.c
int removeElement(int* nums, int numsSize, int val) {
    if (numsSize == 0) return 0;

    // int *newArr[numsSize];
    int k = 0;
    for (int i = 0; i < numsSize; i++)
    {
        if(nums[i] == val)
        {
            continue;
        }else{
            nums[k] = nums[i];
            k++;
        }
    }

    // for (int i = 0; i < k ; i++)
    // {

    //     nums[i] = newArr[i];
    // }

    // Add 1 to last index to return sizeof array
    return k ;
}